### PR TITLE
Issue #588 Port mapping doesn't seem to work

### PR DESCRIPTION
**Pull request by praveenkumar**
This patch need testing against latest build of centos which can be get from http://artifacts.ci.centos.org/minishift/minishift-centos-iso/pr/112/  Once I test it through will remove 'wip' flag.

Addresses issue #588

### CODE REVIEW

**Comment by praveenkumar** [comment]
Before we merge this PR, we should first merge https://github.com/minishift/minishift-centos-iso/pull/112

**Comment by praveenkumar** [comment]
I tested and did work as expected.

**Comment by LalatenduMohanty** [comment]
@praveenkumar is this PR backward compatible with https://github.com/minishift/minishift-centos-iso/releases/tag/v1.0.0-rc.3 ?

**Comment by praveenkumar** [comment]
@LalatenduMohanty It should be, please do a test.

File: pkg/minishift/provisioner/utils.go [hunk 1]
```diff
@@ -19,63 +19,83 @@ package provisioner
...
+Documentation=http://docs.docker.com
 After=network.target rc-local.service
 Requires=rhel-push-plugin.socket
+Requires=docker-cleanup.timer
 
 [Service]
 Type=notify
-ExecStart=/usr/bin/docker daemon -H tcp://0.0.0.0:{{.DockerPort}} -H unix:///var/run/docker.sock \
+NotifyAccess=all
+Environment=GOTRACEBACK=crash
```
Comment (hferentschik) [inline pkg/minishift/provisioner/utils.go]: So how do you know what needs to change? Are you basically copying/updating from a default systemd setup for CentOS?

Have we been using a systemd config which was not in sync with the whatever should have been used from the CentOS side?

**Comment by LalatenduMohanty** [comment]
@praveenkumar I have tested the PR with released CentOS ISO and it seems to work. If you can update the commit message as @hferentschik suggested than that will be perfect.

**Comment by praveenkumar** [comment]
@LalatenduMohanty @hferentschik Updated the commit message.

**Comment by hferentschik** [comment]
Thanks. Testing the Docker reuse case. Once confirmed, I'll merge. Downloads slow though :-(

**Comment by hferentschik** [comment]
Merged, thanks.

**Pull request merged**